src/components/modal/modal.ts

```diff
@@ -3,128 +3,137 @@
-      this._handleOpen();
+  private _elBtnTrigger;
+  private _elBtnClose;
+  private _delay = 400; // ms delay before setting focus/binding events (anim speed of modal open)
+  private _evtOnClickFocusIn = this._onClickFocusIn.bind(this);
+  private _evtOnKeyup = this._onKeyup.bind(this);
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    if (this._isOpen) {
+      this._unbindOpenEvents();
     }
   }
 
-  /**
-   * Allows users to center the modal vertically based on window and modal height.
-   */
-  public centerModal() {
-    const modalBackdrop = this.renderRoot?.querySelector('.' + PREFIX_CLASS + '-modal-backdrop');
-    const modal = this.renderRoot?.querySelector('.' + PREFIX_CLASS + '-modal');
-    const modalHeight = modal.scrollHeight;
-    const windowHeight = window?.innerHeight;
-    const modalMargin = 50-(((100/windowHeight) * modalHeight)/2);
-
-    modalBackdrop.scrollTop = 0;
-    if (windowHeight >= modalHeight && modalMargin > 0) {
-      modal.style.marginTop = 50-(((100/windowHeight) * modalHeight)/2) + 'vh';
-      modal.style.marginBottom = 50-(((100/windowHeight) * modalHeight)/2) + 'vh';
-    }
-    else {
-      modal.style.marginTop = null;
-      modal.style.marginBottom = null;
+  public toggleModal() {
+    if (this._isOpen) {
+      this.closeModal();
+    } else {
+      this.openModal();
     }
   }
 
   @eventOptions({ capture: true, passive: true })
-  private _handleOpen(e) {
-    e?.preventDefault();
-    e?.target.blur();
-    this.centerModal();
-    this.dispatchEvent(new CustomEvent('modal-open'));
+  public openModal() {
+    this._isOpen = true;
+    this._elBtnTrigger = document.activeElement;
+    this._elBtnClose = this.renderRoot.querySelector(`.${PREFIX_CLASS}-modal-control`);
+    setTimeout(() => {
+      this._elBtnClose?.focus();
+      this._bindOpenEvents();
+    }, this._delay);
     document.body.style.overflow = 'hidden';
-    document?.dispatchEvent(new CustomEvent('modal-open'));
-    this._active = true;
   }
 
   @eventOptions({ capture: true, passive: true })
-  private _handleClose(e) {
-    e?.preventDefault();
-    this._active = false;
+  public closeModal() {
+    this._isOpen = false;
     document.body.style.overflow = null;
-    document?.dispatchEvent(new CustomEvent('modal-close'));
-    this.dispatchEvent(new CustomEvent('modal-close'));
+    this._unbindOpenEvents();
+
+    // return focus to trigger
+    if (this._elBtnTrigger) {
+      this._elBtnTrigger.setAttribute('tabindex', '-1');
+      this._elBtnTrigger.focus();
+      setTimeout(() => {
+        this._elBtnTrigger.removeAttribute('tabindex');
+      }, this._delay);
+    }
   }
 
-  private _handleSlotChange() {
-    this.centerModal();
+  @eventOptions({ capture: true })
+  private _onTriggerClick(e) {
+    e.preventDefault();
+    this.openModal();
   }
 
-  connectedCallback() {
-    super.connectedCallback();
-    document?.addEventListener('modal-open', (e) => this._handleClose(e));
```

Comment:
The idea here was to make sure we close any open modals before opening a new one. We're using a document eventListener for this since the modals and triggers aren't technically attached together in any way.

Reply:
This isn't needed since we're trapping focus within the modal. You have to close the modal to access content outside of it.